### DISCUSSION TITLE
Wazuh active response is not working

### DISCUSSION THREAD

**Discussion by andang241** [wazuh/wazuh#24959] [Q&A]
Hello,

I would like to inquire about using the active response feature in Wazuh. I have created a Python file on the Wazuh Windows agent as follows:
```
#!/usr/bin/python3
import os
import sys
import datetime

if os.name == 'nt':
    LOG_FILE = "C:\\Program Files (x86)\\ossec-agent\\active-response\\active-responses.log"
else:
    LOG_FILE = "/var/ossec/logs/active-responses.log"

OS_SUCCESS = 0
OS_INVALID = -1

def write_debug_file(msg):
    with open(LOG_FILE, mode="a") as log_file:
        log_file.write(str(datetime.datetime.now().strftime('%Y/%m/%d %H:%M:%S')) + ": " + msg + "\n")

def main():
    if len(sys.argv) != 2:
        print("Usage: python3 delete_file.py [file_to_delete]")
        sys.exit(OS_INVALID)

    file_to_delete = sys.argv[1]

    write_debug_file("Started delete_file.py")

    try:
        if os.path.exists(file_to_delete):
            os.remove(file_to_delete)
            write_debug_file(f"Successfully removed file: {file_to_delete}")
            print(f"Successfully removed file: {file_to_delete}")
        else:
            write_debug_file(f"File does not exist: {file_to_delete}")
            print(f"File does not exist: {file_to_delete}")
    except OSError as error:
        write_debug_file(f"Error removing file: {file_to_delete} - {error}")
        print(f"Error removing file: {file_to_delete} - {error}")
        sys.exit(OS_INVALID)

    write_debug_file("Ended delete_file.py")
    sys.exit(OS_SUCCESS)

if __name__ == "__main__":
    main()
```
Then, I used PyInstaller to compile the Python file into an .exe file, and then moved that file into the directory C:\Program Files (x86)\ossec-agent\active-response\bin, and finally restarted the Wazuh agent.
On the Wazuh manager, I configured it as follows:
```
<command>
  <name>remove</name>
  <executable>delete_file.exe</executable>
  <timeout_allowed>no</timeout_allowed>
</command>

<active-response>
  <disabled>no</disabled>
  <command>remove</command>
  <location>local</location>
</active-response>
```
After restarting the Wazuh manager, I tried using the Wazuh API to trigger the active response on the Wazuh agent as follows:
![image](https://github.com/user-attachments/assets/d94cc136-6402-4b78-9de8-c94af9cbac8a)
However, it seems that the Wazuh agent is not receiving the arguments. What should I do?

**Comment by MiguelazoDS** [comment]
Hello @andang241, 

If I'm not misunderstanding your script, you're not using stdin to receive the alerts, and that's how active response works since version 4.2.0 https://documentation.wazuh.com/current/release-notes/release-4-2-0.html#breaking-changes 

Please take a look at this template in the documentation 

https://documentation.wazuh.com/current/user-manual/capabilities/active-response/custom-active-response-scripts.html#configuring-the-python-active-response-script-sample

**Comment by sebasfalcone** [comment]
Inactive thread, please reopen if necessary
